src/components/refactor/dialogs/generator/modification/generator-modification-utils.js

```diff
@@ -0,0 +1,279 @@
+            return {
+                [P]: point.p,
+                [Q_MAX_P]: point.qmaxP,
+                [Q_MIN_P]: point.qminP,
+                [PREVIOUS_Q_MAX_P]: point.oldQmaxP ?? point.qmaxP,
+                [PREVIOUS_Q_MIN_P]: point.oldQminP ?? point.qminP,
+                [PREVIOUS_P]: point.oldP ?? point.p,
+            };
+        });
+        const tableData = { [REACTIVE_CAPABILITY_CURVE_TABLE]: rcc };
+        return {
+            ...customEditData,
+            ...tableData,
+        };
+    }
+
+    return {
+        ...customEditData,
+    };
+}
+
+export const assignPreviousValuesToForm = (
+    generator,
+    watchEquipmentId,
+    intl
+) => {
+    // when editing modification form, first render should not trigger this reset
+    // which would empty the form instead of displaying data of existing form
+    let reactiveCapabilityCurvePoints = [
+        getModificationRowEmptyFormData(),
+        getModificationRowEmptyFormData(),
+    ];
+    if (generator?.reactiveCapabilityCurvePoints) {
+        reactiveCapabilityCurvePoints = [];
+    }
+    generator?.reactiveCapabilityCurvePoints?.forEach((element) => {
+        reactiveCapabilityCurvePoints.push({
+            [P]: null,
+            [Q_MIN_P]: null,
+            [Q_MAX_P]: null,
+            [PREVIOUS_P]: element.p ?? null,
+            [PREVIOUS_Q_MIN_P]: element.qminP ?? null,
+            [PREVIOUS_Q_MAX_P]: element.qmaxP ?? null,
+        });
+    });
+    const energySourceLabelId = getEnergySourceLabel(generator?.energySource);
+    const previousEnergySourceLabel = energySourceLabelId
+        ? intl.formatMessage({
+              id: energySourceLabelId,
+          })
+        : undefined;
+    const previousVoltageRegulationType = generator?.voltageRegulatorOn
+        ? generator?.regulatingTerminalVlId ||
+          generator?.regulatingTerminalConnectableId
+            ? intl.formatMessage({
+                  id: REGULATION_TYPES.DISTANT.label,
+              })
+            : intl.formatMessage({ id: REGULATION_TYPES.LOCAL.label })
+        : null;
+
+    const previousFrequencyRegulationState = getPreviousBooleanValue(
+        generator?.activePowerControlOn
+    );
+
+    const previousEquipment =
+        generator?.regulatingTerminalConnectableId &&
+        generator?.regulatingTerminalConnectableType
+            ? generator?.regulatingTerminalConnectableType +
+              ':' +
+              generator?.regulatingTerminalConnectableId
+            : null;
+    return {
+        [EQUIPMENT_ID]: watchEquipmentId,
+        [REACTIVE_CAPABILITY_CURVE_TABLE]: reactiveCapabilityCurvePoints,
+        [REACTIVE_CAPABILITY_CURVE_CHOICE]:
+            generator?.minMaxReactiveLimits != null ? 'MINMAX' : 'CURVE',
+        [VOLTAGE_REGULATION]: generator?.voltageRegulatorOn
+            ? generator?.voltageRegulatorOn
+            : null,
+        [FREQUENCY_REGULATION]: generator?.activePowerControlOn,
```

Comment:
Remove [VOLTAGE_REGULATION] and [FREQUENCY_REGULATION] so they can be null by default when selecting an new id